sql/core/src/main/scala/org/apache/spark/sql/api.scala

```diff
@@ -171,6 +171,33 @@ private[sql] trait DataFrameSpecificApi {
   def saveAsTable(tableName: String): Unit
 
   @Experimental
+  def saveAsTable(tableName: String, dataSourceName: String, options: (String, String)*): Unit
+
+  @Experimental
+  def saveAsTable(
+      tableName: String,
+      dataSourceName: String,
+      options: java.util.Map[String, String]): Unit
+
+  @Experimental
+  def save(path: String): Unit
+
+  @Experimental
+  def save(path: String, overwrite: Boolean): Unit
+
+  @Experimental
+  def save(
+      dataSourceName: String,
+      overwrite: Boolean,
+      options: (String, String)*): Unit
```

Comment:
i think this one might be ambiguous with the other save.

u'd want to declare something like
```scala
def save(
      dataSourceName: String,
      overwrite: Boolean,
      option: (String, String),
      options: (String, String)*): Unit
```